packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -120,6 +174,55 @@ export default class SaytDriverPlugin implements Plugin {
     return [searchTerms];
   }
 
+  /**
+   * Extracts query and products from the given response.
+   *
+   * @param response An object containing the original query and product records.
+   * @returns An object containing the query and an array of valid simplified products.
+   */
+  searchCallback(response: Results): ProductsResponseSection {
+    const { query, records } = response;
+    const mappedRecords = records.map(record => {
+      const filter = this.parseRecord(record);
+      if (filter === undefined) return;
+      const { data, firstVariant, nonvisualVariants } = filter;
+      return {
+        title: data.title,
+        price: nonvisualVariants[0].originalPrice,
+        imageSrc: firstVariant.productImage,
+        imageAlt: data.title,
+        productUrl: firstVariant.productImage,
+        // @TODO Handle variants
+      }
+    }).filter(record => record);
```

Comment:
Is `.filter(Boolean)` a common enough idiom for us to use?

Reply:
Clever. I think the meaning would be pretty clear, even if people haven't seen it done before.